src/addons/presenter.js

```diff
@@ -248,33 +251,44 @@ class PresenterMediator extends React.PureComponent {
     return this.model.value
   }
 
+  // Do not allow transfer across repos. Check to for inheritence by comparing
+  // the common history object shared between repos
   hasParent(): boolean {
-    // Do not allow transfer across repos. Check to for inheritence by comparing
-    // the common history object shared between repos
+    // No need to check if there isn't a parent context setup
+    if (!this.context.parent) {
+      return false
+    }
+
     return get(this.repo, 'history') === get(this.context, ['repo', 'history'])
   }
```

Comment:
> my understanding is that the only problem this is solving is the fact that an action should be caught by the closest repo available

Precisely.

> so you're using history as a unique identifier of sorts? It's confusing to me to see history used in comparison here

Yes. We just need to prevent parent apps from stealing the actions of children. We may be able to tell by checking for the `repo` prop instead.

> doesn't seem like it has a logical reason to be in this area of code.

What do you recommend?